File: doc-examples/nitrite-example-java/src/main/java/example/CompositeFkChildRepository.java
package example;

import io.micronaut.data.annotation.Join;
import io.micronaut.data.nitrite.annotation.NitriteRepository;
import io.micronaut.data.repository.CrudRepository;

import java.util.Optional;

// tag::composite-fk-repository[]
@NitriteRepository
public interface CompositeFkChildRepository extends CrudRepository<CompositeFkChild, String> {
    @Join("parent")
    Optional<CompositeFkChild> findByName(String name);
}
// end::composite-fk-repository[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/CompositeFkChildRepository.kt
package example

import io.micronaut.data.annotation.Join
import io.micronaut.data.nitrite.annotation.NitriteRepository
import io.micronaut.data.repository.CrudRepository
import java.util.Optional

// tag::composite-fk-repository[]
@NitriteRepository
interface CompositeFkChildRepository : CrudRepository<CompositeFkChild, String> {
    @Join("parent")
    fun findByName(name: String): Optional<CompositeFkChild>
}
// end::composite-fk-repository[]
